whisper_experiments/utils/string_utils.py

```diff
@@ -0,0 +1,329 @@
+import rapidfuzz
+import text_diff
+from typing import (
+    NamedTuple,
+    Union,
+    Callable,
+    Optional,
+    List,
+    Iterable,
+    Iterator,
+    Tuple,
+)
+from text_diff import UnchangedLine, ModifiedLine, RemovedLine, AddedLine
+from itertools import filterfalse
+
+# Aliases to just indicate the object refers to a word
+ModifiedWord = ModifiedLine
+RemovedWord = RemovedLine
+AddedWord = AddedLine
+Word = Union[ModifiedWord, RemovedWord, AddedWord]
+Line = Union[ModifiedLine, RemovedLine, AddedLine]
+TextType = Union[Word, Line]
+
+SimilarityCalculator = Callable[[str, str], float]
+WordSplitter = Callable[[str], Iterable[str]]
+
+
+class TextDiff:
+    """
+    Wrapper for text_diff objects e.g. ModifiedLine.
+    """
+
+    def __init__(self, text_diff: TextType):
+        """
+        Parameters
+        ----------
+        text_diff: TextType
+            The text_diff object to wrap
+        """
+        self.text_diff = text_diff
+
+    def __eq__(self, other) -> bool:
+        return (
+            self.is_removed == other.is_removed
+            and self.is_added == other.is_added
+            and self.is_modified == other.is_modified
+            and self.content == other.content
+            and self.content_before == other.content_before
+            and self.content_after == other.content_after
+        )
+
+    def __str__(self) -> str:
+        if self.is_modified:
+            return f"Modified: {self.content_before} -> {self.content_after}"
+        if self.is_removed:
+            return f"Removed: {self.content}"
+        return f"Added: {self.content}"
```

Comment:
Gonna replace all this with just `str(self.text_diff)` in some subsequent PR. i.e. Just return the underlying `text_diff` object's `str` representation.